### PR TITLE
set volume/muted when ima throws volume change events (needs to be do…

### PR DESCRIPTION
**Pull request by evaschmuhl**
…ne before video starts playing)

The ima sends some events when the volume changes. The default ui needs to listen to them to set the muted state before the content video resumes to play.
Setting those values from the local storage on the play event listener would be too late and the content video would be paused directly after muting/unmuting.

Sandbox: https://esc-1.sb.familie.de/schwangerschaft/vornamen/alte-vornamen/

### CODE REVIEW

File: src/UI.ts [hunk 1]
```diff
@@ -61,6 +62,7 @@ class UI {
     this.onVideoElTimeupdate = noop
     this.onVideoElVolumeChange = noop
     this.onLoadedMetaData = noop
+    this.onVolumeChange = noop
```
Comment (jrsmb) [inline src/UI.ts:65]: The naming confuses me. Why are there two functions called
`onVideoElVolumeChange` and `onVolumeChange`? Maybe add `ima` to the name somehow if it only relates to that

File: src/UI.ts [hunk 2]
```diff
@@ -629,6 +631,19 @@ class UI {
     }
     videoEl.addEventListener('play', this.onVideoElPlay)
 
+    this.onVolumeChange = (event: any) => {
+      Logger.log('ima volume change ')
+      videoEl.muted = convertLocalStorageIntegerToBoolean('StroeerVideoplayerMuted')
+      videoEl.volume = convertLocalStorageStringToNumber('StroeerVideoplayerVolume')
+      const volumeInPercent = videoEl.volume * 100
+      volumeLevelBubble.style.bottom = 'calc(' + String(volumeInPercent) + '% - 4px)'
+      volumeLevelBubble.style.top = 'auto'
```
Comment (jrsmb) [inline src/UI.ts:640]: Could this setting be moved to where the initial settings of the volume bar are set? This seems to have a constant value which does not need to be re-set

**Review by jrsmb — COMMENTED** [review]
Makes sense, I like this solution :)

**Review by ennoStroeer — APPROVED** [review]

File: src/UI.ts [hunk 3]
```diff
@@ -629,6 +631,19 @@ class UI {
     }
     videoEl.addEventListener('play', this.onVideoElPlay)
 
+    this.onImaVolumeChange = (event: any) => {
+      Logger.log('ima volume change ')
```
Comment (jrsmb) [inline src/UI.ts:635]: Not sure if you meant to keep this in. Otherwise clean!

**Review by jrsmb — APPROVED** [review]

**Pull request merged**